src/index.ts

```diff
@@ -112,45 +157,126 @@ export const createInstance = (config: Config): Instance => {
         ? config.zAuctionLegacyAddress
         : config.zAuctionAddress;
 
-      const erc20Token = await getZAuctionTradeToken(
-        config.web3Provider,
-        zAuctionAddress
-      );
+      const bidToken = bid.bidToken ?? config.wildTokenAddress;
 
-      const allowance = await actions.getZAuctionTradeTokenAllowance(
+      const allowance = await actions.getPaymentTokenAllowance(
         account,
-        zAuctionAddress,
-        erc20Token
+        bidToken,
+        config.web3Provider,
+        zAuctionAddress
       );
 
       return allowance;
     },
-
     getZAuctionSpendAllowance: async (
+      paymentTokenAddress: string,
       account: string
     ): Promise<ethers.BigNumber> => {
-      const erc20Token = await getZAuctionTradeToken(
+      const allowance = await actions.getPaymentTokenAllowance(
+        account,
+        paymentTokenAddress,
         config.web3Provider,
         config.zAuctionAddress
       );
 
-      const allowance = await actions.getZAuctionTradeTokenAllowance(
+      return allowance;
+    },
+    getZAuctionSpendAllowanceLegacy: async (
+      account: string
+    ): Promise<ethers.BigNumber> => {
+      const allowance = await actions.getPaymentTokenAllowance(
         account,
-        config.zAuctionAddress,
-        erc20Token
+        config.wildTokenAddress,
+        config.web3Provider,
+        config.zAuctionAddress
       );
 
       return allowance;
     },
-    getTradeTokenAddress: async (): Promise<string> => {
-      const erc20Token = await getZAuctionTradeToken(
+    // Return the default ERC20 token specified by zAuction
+    getDefaultPaymentToken: async (): Promise<string> => {
+      const contract = await getZAuctionContract(
         config.web3Provider,
         config.zAuctionAddress
       );
+      const defaultToken = await contract.defaultPaymentToken();
+      return defaultToken;
+    },
+    // Set the default ERC20 token for zAuction to use
+    setDefaultPaymentToken: async (
+      defaultTokenAddress: string,
+      signer: ethers.Signer
+    ): Promise<ethers.ContractTransaction> => {
+      const contract = await getZAuctionContract(
+        config.web3Provider,
+        config.zAuctionAddress
+      );
+
+      const zAuctionOwner = await contract.owner();
+      const signerAddress = await signer.getAddress();
 
-      return erc20Token.address;
+      if (signerAddress !== zAuctionOwner) {
+        throw Error(
+          "Cannot set the default token for zAuction if you are not the owner"
+        );
+      }
+
+      const tx = await contract
+        .connect(signer)
+        .setDefaultPaymentToken(defaultTokenAddress);
+      return tx;
     },
+    // Return the ERC20 token specified as the payment token for a network.
+    // If not set, or is intentionally 0, it will return the default token
+    getDomainNetworkPaymentToken: async (
```

Comment:
If we expose this function, it should only accept `networkId`'s which are actually network domains, and throw an error if it's not a network.

Reply:
How would we enforce that a domain must be a network domain?

Reply:
You could call the `.parentOf` on ZNS Hub and check if the parent is the zero hash